packages/protocol/contracts/common/AuthorizationBase.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.20;
+
+import { EssentialContract } from "../common/EssentialContract.sol";
+
+/// @title AuthorizationBase
+/// @notice Every contract which needs to have authorization (EtherVault, ERCXXVault)
+abstract contract AuthorizationBase {
```

Comment:
is EssentialContract

Reply:
Tried the idea, but it would create unambigous parent dependencies (EtherVault vs. ERCXXXVault)

Reply:
Ot making baseNFTVault ingeriting from that one can do the job. ok, will try later.

Reply:
> Or making baseNFTVault ingeriting from that one can do the job. ok, will try later.

If possile, this is the preferred approach.